File: Sorting/442.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
  let i = 0;
  let result = [];
  
  while(i < nums.length){
    currNum = nums[i];
    if(currNum != nums[currNum - 1]){
      let temp = nums[currNum - 1];
      nums[currNum - 1] = nums[i];
      nums[i] = temp;
    } else {
      i++;
    }
  }

  for(let j = 0; j < nums.length; j++){
    let currNum = nums[j];
    if(currNum != j + 1){
      result.push(currNum);
    }
  }

  return result;
};

File: Sorting/442.cs
public class Solution {
    public IList<int> FindDuplicates(int[] nums) {
      List<int> result = new List<int>();          
      int i = 0;

      while(i < nums.Length)
      {
        int currNum = nums[i];
        if(currNum != nums[currNum - 1])
        {
          //swap
          int temp = nums[currNum - 1];
          nums[currNum - 1] = nums[i];
          nums[i] = temp;
        }
        else
        {
          i++; 
        }
      }

      for(int j = 0; j < nums.Length; j++)
      {
        int currNum = nums[j];
        if(currNum != j + 1)   
        {
          result.Add(currNum);
        }
      }

      return result;
    }
}